src/ExamsPage/hooks.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+
+import { RequestKeys } from 'data/constants';
+
+import * as reduxHooks from 'data/redux/hooks';
+
+import * as api from './data/api';
+import { loadExams } from './data/reducer';
+
+export const useFetchCourseExams = () => {
+  const makeNetworkRequest = reduxHooks.useMakeNetworkRequest();
+  const dispatch = useDispatch();
+  return (courseId) => (
+    makeNetworkRequest({
+      requestKey: RequestKeys.fetchCourseExams,
+      promise: api.getCourseExams(courseId),
+      onSuccess: (response) => dispatch(loadExams(response)),
+    })
+  );
+};
+
+export const useInitializeExamsPage = (courseId) => {
+  const fetchCourseExams = useFetchCourseExams();
+  React.useEffect(() => { fetchCourseExams(courseId); }, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
I notice you have an `eslint-disable-line` here for the `react-hooks/exhaustive-deps` rule. `fetchCourseExams` and `courseId` should be in the dependency array, no? I imagine that the reason you're not doing this is because the `fetchCourseExams` reference will change on every call to `useFetchCourseExams` and you want to avoid network calls on every re-render of the page. Is that right? If so, I think a way to get around this is to use the [useCallback](https://react.dev/reference/react/useCallback) hook to cache the reference to the lambda function returned by `useFetchCourseExams`. That way, you can include `fetchCourseExams` in the dependency array without causing a network call on every re-render.

Reply:
that's basically the idea, I wanted this to run only once (we don't expect course id to change either). I did some looking into useCallback and it seems like it's recommended to avoid unless you're solving a specific performance issue since it creates a bunch of overhead. I kinda just copied this pattern from the learner-dashboard for creating an initialize hook but there might be other ways to accomplish this.